File: core/src/main/java/com/victorlh/android/framework/core/dialog/AbstractDialogFragment.java
package com.victorlh.android.framework.core.dialog;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.victorlh.android.framework.core.fragment.AbstractFragment;
import com.victorlh.android.framework.core.fragment.IEventoFragment;

/**
 * @author Victor
 * 09/06/2018
 */
public abstract class AbstractDialogFragment extends DialogFragment implements IDialog, IEventoFragment {

	private int viewId;
	protected View view;

	private AbstractFragment fragment;
	private FragmentManager fragmentManager;

	protected EventosDialogs.OnCancelListener onCancelListener;
	private EventosDialogs.OnDissmissListener onDissmissListener;

	private int height;
	private int width;

	public AbstractDialogFragment() {
		this.width = ViewGroup.LayoutParams.MATCH_PARENT;
		this.height = ViewGroup.LayoutParams.MATCH_PARENT;
	}

	public void init(int viewId, AbstractFragment fragment, FragmentManager fragmentManager) {
		this.viewId = viewId;
		this.fragment = fragment;
		this.fragmentManager = fragmentManager;
	}

	@Override
	public void onSaveInstanceState(@NonNull Bundle outState) {
		super.onSaveInstanceState(outState);
		outState.putInt("viewId", viewId);
	}

	@Override
	public void onViewStateRestored(@Nullable Bundle savedInstanceState) {
		super.onViewStateRestored(savedInstanceState);
		if (savedInstanceState != null) {
			this.viewId = savedInstanceState.getInt("viewId");
		}
	}

	@Nullable
	@Override
	public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
		if (viewId == 0) {
			return null;
		}
		this.view = inflater.inflate(viewId, container, false);

		if (fragment != null) {
			fragment.setLanzadorEvento(this);
			FragmentManager childFragmentManager = getChildFragmentManager();
			FragmentTransaction fragmentTransaction = childFragmentManager.beginTransaction();
			fragmentTransaction.add(getIdLyFragmentBody(), fragment);
			fragmentTransaction.commit();
		}

		return view;
	}

	@Override
	public void onStart() {
		super.onStart();

		Dialog dialog = getDialog();
		if (dialog != null) {
			Window window = dialog.getWindow();
			if (window != null) {
				window.setLayout(width, height);
			}

			if (onCancelListener != null) {
				dialog.setOnCancelListener(new DialogInterface.OnCancelListener() {
					@Override
					public void onCancel(DialogInterface dialog) {
						if (onCancelListener != null) {
							onCancelListener.onCancel(AbstractDialogFragment.this);
						}
					}
				});
				dialog.setOnDismissListener(new DialogInterface.OnDismissListener() {
					@Override
					public void onDismiss(DialogInterface dialog) {
						if (onCancelListener != null) {
							onCancelListener.onCancel(AbstractDialogFragment.this);
						}
					}
				});
			}
		}
	}

	protected abstract int getIdLyFragmentBody();

	@Override
	public void show() {
		if (fragmentManager != null) {
			String tag = fragment == null ? "" : fragment.getClass().getName();
			show(fragmentManager, tag);
		}
	}

	@Override
	public void close() {
		dismiss();
	}

	@Override
	public void setOnCancelListener(EventosDialogs.OnCancelListener onCancelListener) {
		this.onCancelListener = onCancelListener;
	}

	@Override
	public void setOnDissmissListener(EventosDialogs.OnDissmissListener onDissmissListener) {
		this.onDissmissListener = onDissmissListener;
	}

	public void setHeight(int height) {
		this.height = height;
	}

	public void setWidth(int width) {
		this.width = width;
	}
}

File: core/src/main/java/com/victorlh/android/framework/core/dialog/AbstractDialogFragment.kt
package com.victorlh.android.framework.core.dialog

import android.content.DialogInterface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.FragmentManager
import com.victorlh.android.framework.core.fragment.AbstractFragment

/**
 * @author Victor
 * 09/06/2018
 */
abstract class AbstractDialogFragment : DialogFragment(), IDialog {

	private var viewId = 0
	protected lateinit var fragment: AbstractFragment
	private lateinit var mFragmentManager: FragmentManager

	protected var viewContainer: View? = null
	private var onCancelListener: ((dialog: IDialog) -> Unit)? = null
	private var onDissmissListener: (() -> Unit)? = null

	private var height: Int = ViewGroup.LayoutParams.MATCH_PARENT
	private var width: Int = ViewGroup.LayoutParams.MATCH_PARENT

	fun init(viewId: Int, fragment: AbstractFragment, fragmentManager: FragmentManager) {
		this.viewId = viewId
		this.fragment = fragment
		this.mFragmentManager = fragmentManager
	}

	override fun onSaveInstanceState(outState: Bundle) {
		super.onSaveInstanceState(outState)
		outState.putInt("viewId", viewId)
	}

	override fun onViewStateRestored(savedInstanceState: Bundle?) {
		super.onViewStateRestored(savedInstanceState)
		if (savedInstanceState != null) {
			viewId = savedInstanceState.getInt("viewId")
		}
	}

	override fun onCreateView(
			inflater: LayoutInflater,
			container: ViewGroup?,
			savedInstanceState: Bundle?
	): View? {
		if (viewId == 0) {
			return null
		}
		return inflater.inflate(viewId, container, false)
	}

	override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
		super.onViewCreated(view, savedInstanceState)

		val childFragmentManager = childFragmentManager
		val fragmentTransaction = childFragmentManager.beginTransaction()
		fragmentTransaction.add(idLyFragmentBody, fragment)
		fragmentTransaction.commit()
	}

	override fun onStart() {
		super.onStart()
		val dialog = dialog
		if (dialog != null) {
			val window = dialog.window
			window?.setLayout(width, height)
			if (onCancelListener != null) {
				dialog.setOnCancelListener(DialogInterface.OnCancelListener {
					onCancelListener?.invoke(this@AbstractDialogFragment)
				})
				dialog.setOnDismissListener(DialogInterface.OnDismissListener {
					onDissmissListener?.invoke()
				})
			}
		}
	}

	protected abstract val idLyFragmentBody: Int

	override fun show() {
		val tag = fragment.javaClass.name
		show(mFragmentManager, tag)
	}

	override fun close() {
		dismiss()
	}

	override fun setOnCancelListener(evento: (dialog: IDialog) -> Unit) {
		this.onCancelListener = evento
	}

	override fun setOnDissmissListener(evento: () -> Unit) {
		this.onDissmissListener = evento
	}

	fun setHeight(height: Int) {
		this.height = height
	}

	fun setWidth(width: Int) {
		this.width = width
	}
}